examples/vllm_inference/flow.py

```diff
@@ -1,34 +1,35 @@
 from metaflow import (
     FlowSpec,
+    Parameter,
     step,
-    card,
     pypi,
     kubernetes,
-    parallel,
-    secrets,
+    card,
     huggingface_hub,
-    current,
     model,
-    metaflow_ray,
+    JSONType,
+    current,
+    metaflow_ray
 )
 
-DISK_SIZE = 100 * 1000  # 100 GB
-
-MEMORY = 60 * 1000  # 60 GB
 
+class vLLMMistralInference(FlowSpec):
+    messages = Parameter(
+        name="messages",
+        type=JSONType,
+        required=True,
+        help="messages in json format"
+    )
 
-class TestLLM(FlowSpec):
     @step
     def start(self):
         self.next(self.pull_model_from_huggingface)
 
     @pypi(
-        python="3.10.11",
+        python="3.10",
         packages={
-            "vllm": "0.6.1",
-            "transformers": "4.44.2",
-            "huggingface-hub": "0.25.1",
-        },
+            "huggingface-hub": "0.27.0",
+        }
```

Comment:
You can even parameterize the `self.model_id`. If you want to play with free OSS models, you can use the `unsloth/Llama-3.2-3B-Instruct` instead of this one to test the generation bit. It should work on the infra deployed in `dev-valay`.